great_expectations/rule_based_profiler/parameter_builder/metric_parameter_builder.py

```diff
@@ -72,7 +73,16 @@ def _build_parameters(
         )
 
         # Obtain domain kwargs from rule state (i.e., variables and parameters); from instance variable otherwise.
-        metric_domain_kwargs: Optional[
+        domain_kwargs: dict = get_parameter_value_and_validate_return_type(
+            domain=domain,
+            parameter_reference="$domain.domain_kwargs",
+            expected_return_type=dict,
+            variables=variables,
+            parameters=parameters,
+        )
+
+        metric_domain_kwargs: dict = copy.deepcopy(domain_kwargs)
+        metric_domain_kwargs_override: Optional[
```

Comment:
Is it possible to identify whether or not the user is overriding without having to reinvoke `get_parameter_value_and_validate_return_type`? 

Is the override dependent on the value of `self._metric_domain_kwargs`? If so, can we simplify this with a conditional like
```
if self._metric_domain_kwargs:
    do override 
else: 
    do standard
```
to prevent having to copy and reinvoke that method? Let me know if my understanding is off here.

Reply:
@cdkini To your point, I have been thinking of how to improve this; not in this PR, but in a cleanup PR we can encapsulate the override logic into a common utility so that it is easy to use and less error prone.  But for now, I feel that we do not need to make changes.

Reply:
@cdkini This was actually done now!